src/ApiIncrementalChecker.ts

```diff
@@ -165,4 +193,25 @@ export class ApiIncrementalChecker implements IncrementalCheckerInterface {
       allLints.map(this.createNormalizedMessageFromRuleFailure)
     );
   }
+
+  public getEsLints(cancellationToken: CancellationToken) {
+    cancellationToken.throwIfCancellationRequested();
```

Comment:
I think this line should be inside `for (const updatedFile of this.lastUpdatedFiles)` on line 205 (before `if` statement)

Reply:
I don't think so - the value that `cancellationToken.throwIfCancellationRequested` depends upon can't change mid loop iteration. At least as I understand it.

Unless there's something I've missed?

Reply:
The thing is that CancellationToken uses synchronous file system calls - the file system is a very simple/primitive way to communicate between processes. Thanks to that we can cancel type-checking/linting in the middle of the process (which is basically a loop over files). The idea of file-base cancellation token came from typescript implementation of CancellationToken :)

Reply:
FIxed!